### ISSUE TITLE
Rails routes outputs all of flippers rack stack

### ISSUE THREAD

**Issue by IanVaughan** [jnunemaker/flipper#614]
Rails routes for flipper used to be just the Flipper:UI
```
/flipper                                                      Flipper::UI
```

But now its a massive stack trace of all the rack chain
```
$ rails routes -E -g flipper
--[ Route 1 ]---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Prefix            |
Verb              |
URI               | /flipper
Controller#Action | #<Rack::Protection::AuthenticityToken:0x00007fc3ff6ebc78 @app=#<Rack::Protection::FrameOptions:0x00007fc3ff6ebde0 @app=#<Rack::Protection::HttpOrigin:0x00007fc3ff6ebf48 @app=#<Rack::Protection::IPSpoofing:0x00007fc3ff6e00a8 @app=#<Rack::Protection::JsonCsrf:0x00007fc3ff6e0210 @app=#<Rack::Protection::PathTraversal:0x00007fc3ff6e0350 @app=#<Rack::Protection::RemoteToken:0x00007fc3ff6e04b8 @app=#<Rack::Protection::SessionHijacking:0x00007fc3ff6e0620 @app=#<Rack::Protection::XSSHeader:0x00007fc3ff6e07d8 @app=#<Rack::MethodOverride:0x00007fc4026495b0 @app=#<Flipper::Middleware::SetupEnv:0x00007fc4026496a0 @app=#<Flipper::UI::Middleware:0x00007fc402649a38 @app=#<Proc:0x00007fc40264a140 /Users/ianvaughan/.asdf/installs/ruby/3.0.3/lib/ruby/gems/3.0.0/gems/flipper-ui-0.24.0/lib/flipper/ui.rb:43 (lambda)>, @env_key="flipper", @flipper=Flipper, @action_collection=#<Flipper::UI::ActionCollection:0x00007fc402649970 @action_classes=[Flipper::UI::Actions::AddFeature, Flipper::UI::Actions::ActorsGate, Flipper::UI::Actions::GroupsGate, Flipper::UI::Actions::BooleanGate, Flipper::UI::Actions::PercentageOfTimeGate, Flipper::UI::Actions::PercentageOfActorsGate, Flipper::UI::Actions::Feature, Flipper::UI::Actions::Features, Flipper::UI::Actions::File, Flipper::UI::Actions::Home]>>, @env_key="flipper", @flipper=Flipper>>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :xss_mode=>:block, :nosniff=>true, :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :tracking_key=>:tracking, :encrypt_tracking=>true, :track=>["HTTP_USER_AGENT"], :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :authenticity_param=>"authenticity_token", :allow_if=>nil, :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :allow_if=>nil, :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :allow_if=>nil, :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :frame_options=>:sameorigin, :use=>:authenticity_token}>, @options={:reaction=>:default_reaction, :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1, :session_key=>"rack.session", :status=>403, :allow_empty_referrer=>true, :report_key=>"protection.failed", :html_types=>["text/html", "application/xhtml", "text/xml", "application/xml"], :authenticity_param=>"authenticity_token", :allow_if=>nil, :use=>:authenticity_token}>
```
Which keeps changing on each rails routes request, I guess the order changes randomly?

We use rails routes output to monitor changes in requests, of which this is effecting.

**Comment by jnunemaker** [comment]
What version of rails/flipper/flipper-ui are you using?

On March 16, 2022, David Grevink ***@***.***> wrote:
> Rails routes for flipper used to be just the Flipper:UI
>
> /flipper Flipper::UI
> But now its a massive stack trace of all the rack chain
>
> $ rails routes -E -g flipper --[ Route 1
> ]---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
> Prefix | Verb | URI | /flipper Controller#Action |
> #<Rack::Protection::AuthenticityToken:0x00007fc3ff6ebc78
> @app=#<Rack::Protection::FrameOptions:0x00007fc3ff6ebde0
> @app=#<Rack::Protection::HttpOrigin:0x00007fc3ff6ebf48
> @app=#<Rack::Protection::IPSpoofing:0x00007fc3ff6e00a8
> @app=#<Rack::Protection::JsonCsrf:0x00007fc3ff6e0210
> @app=#<Rack::Protection::PathTraversal:0x00007fc3ff6e0350
> @app=#<Rack::Protection::RemoteToken:0x00007fc3ff6e04b8
> @app=#<Rack::Protection::SessionHijacking:0x00007fc3ff6e0620
> @app=#<Rack::Protection::XSSHeader:0x00007fc3ff6e07d8
> @app=#<Rack::MethodOverride:0x00007fc4026495b0
> @app=#<Flipper::Middleware::SetupEnv:0x00007fc4026496a0
> @app=#<Flipper::UI::Middleware:0x00007fc402649a38
> @app=#<Proc:0x00007fc40264a140
> /Users/ianvaughan/.asdf/installs/ruby/3.0.3/lib/ruby/gems/3.0.0/gems/flipper-
> ui-0.24.0/lib/flipper/ui.rb:43 (lambda)>, @env_key="flipper",
> @flipper=Flipper,
> @action_collection=#<Flipper::UI::ActionCollection:0x00007fc402649970
> @action_classes=[Flipper::UI::Actions::AddFeature,
> Flipper::UI::Actions::ActorsGate, Flipper::UI::Actions::GroupsGate,
> Flipper::UI::Actions::BooleanGate,
> Flipper::UI::Actions::PercentageOfTimeGate,
> Flipper::UI::Actions::PercentageOfActorsGate,
> Flipper::UI::Actions::Feature, Flipper::UI::Actions::Features,
> Flipper::UI::Actions::File, Flipper::UI::Actions::Home]>>,
> @env_key="flipper", @flipper=Flipper>>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :xss_mode=>:block, :nosniff=>true,
> :use=>:authenticity_token}>, @options={:reaction=>:default_reaction,
> :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :tracking_key=>:tracking, :encrypt_tracking=>true,
> :track=>["HTTP_USER_AGENT"], :use=>:authenticity_token}>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :authenticity_param=>"authenticity_token",
> :allow_if=>nil, :use=>:authenticity_token}>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :use=>:authenticity_token}>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :allow_if=>nil, :use=>:authenticity_token}>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :use=>:authenticity_token}>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :allow_if=>nil, :use=>:authenticity_token}>,
> @options={:reaction=>:default_reaction, :logging=>true,
> :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :frame_options=>:sameorigin,
> :use=>:authenticity_token}>, @options={:reaction=>:default_reaction,
> :logging=>true, :message=>"Forbidden", :encryptor=>Digest::SHA1,
> :session_key=>"rack.session", :status=>403,
> :allow_empty_referrer=>true, :report_key=>"protection.failed",
> :html_types=>["text/html", "application/xhtml", "text/xml",
> "application/xml"], :authenticity_param=>"authenticity_token",
> :allow_if=>nil, :use=>:authenticity_token}>
> Which keeps changing on each rails routes request, I guess the order
> changes randomly?
>
> We use rails routes output to monitor changes in requests, of which
> this is effecting.
>

—
Reply to this email directly, view it on GitHub
<https://github.com/jnunemaker/flipper/issues/614>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AAAAB2ZLMLY4J6VFTLQXU6DVAHIEBANCNFSM5Q3XOAGQ>.
You are receiving this because you are subscribed to this thread.Message
ID: ***@***.***>

**Comment by jnunemaker** [comment]
@IanVaughan what version of rails though?

**Comment by rafaeelaudibert** [comment]
I'm also seeing the same with Rails 4.2

<img width="1250" alt="image" src="https://user-images.githubusercontent.com/32079912/159406935-7cd65105-fa91-4bff-8eec-302d9c2f93fd.png">
> Sorry for low resolution, only way to make it fit in my small laptop screen

**Comment by IanVaughan** [comment]
* rails (6.1.4.6)

**Comment by jnunemaker** [comment]
I have this all fixed up in #619. Should be merged and out shortly. Sorry for the regression.
